# Exploration du fichier source 'title.basics.tsv'

In [ ]:
# Importer le fichier source 'title.basics.tsv' dans un dataframe Pandas : 

import pandas as pd
df = pd.read_csv("C:/Users/vieas/Documents/GitHub/Movie-recommendation-project-with-Wild-Code-School/gitignore/title.basics.tsv",sep='\t')

C:\Users\vieas\AppData\Local\Temp\ipykernel_6232\1430891619.py:4: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("C:/Users/vieas/Documents/GitHub/Movie-recommendation-project-with-Wild-Code-School/gitignore/title.basics.tsv",sep='\t')


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11248099 entries, 0 to 11248098
Data columns (total 9 columns):
 #   Column          Dtype 
---  ------          ----- 
 0   tconst          object
 1   titleType       object
 2   primaryTitle    object
 3   originalTitle   object
 4   isAdult         object
 5   startYear       object
 6   endYear         object
 7   runtimeMinutes  object
 8   genres          object
dtypes: object(9)
memory usage: 772.3+ MB


In [6]:
# Obtenir un apperçu des premières et dernières lignes + obtenir les lignes centrales ou aléatoires : 
display(df.head())
display(df.tail())

tconst titleType            primaryTitle           originalTitle  \
0  tt0000001     short              Carmencita              Carmencita   
1  tt0000002     short  Le clown et ses chiens  Le clown et ses chiens   
2  tt0000003     short            Poor Pierrot          Pauvre Pierrot   
3  tt0000004     short             Un bon bock             Un bon bock   
4  tt0000005     short        Blacksmith Scene        Blacksmith Scene   

  isAdult startYear endYear runtimeMinutes                    genres  
0       0      1894      \N              1         Documentary,Short  
1       0      1892      \N              5           Animation,Short  
2       0      1892      \N              5  Animation,Comedy,Romance  
3       0      1892      \N             12           Animation,Short  
4       0      1893      \N              1              Comedy,Short

tconst  titleType               primaryTitle  \
11248094  tt9916848  tvEpisode              Episode #3.17   
11248095  tt9916850  tvEpisode              Episode #3.19   
11248096  tt9916852  tvEpisode              Episode #3.20   
11248097  tt9916856      short                   The Wind   
11248098  tt9916880  tvEpisode  Horrid Henry Knows It All   

                      originalTitle isAdult startYear endYear runtimeMinutes  \
11248094              Episode #3.17       0      2009      \N             \N   
11248095              Episode #3.19       0      2010      \N             \N   
11248096              Episode #3.20       0      2010      \N             \N   
11248097                   The Wind       0      2015      \N             27   
11248098  Horrid Henry Knows It All       0      2014      \N             10   

                              genres  
11248094         Action,Drama,Family  
11248095         Action,Drama,Family  
11248096         Action,Drama,Family  
11248097                       Short  
11248098  Adventure,Animation,Comedy

In [5]:
# Obtenir les informations principales : 
df.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11248099 entries, 0 to 11248098
Data columns (total 9 columns):
 #   Column          Dtype 
---  ------          ----- 
 0   tconst          object
 1   titleType       object
 2   primaryTitle    object
 3   originalTitle   object
 4   isAdult         object
 5   startYear       object
 6   endYear         object
 7   runtimeMinutes  object
 8   genres          object
dtypes: object(9)
memory usage: 772.3+ MB


# Analyse à partir de du résultat de 'df.head()', 'df.tail()' et 'df.info()' : 

11248098 lignes
9 colonnes
poids = 772.3+ MB

#   Column          Dtype 
---  ------          ----- 
 0   tconst          object --> censé être un string, donc ok pour le format
 1   titleType       object --> censé être un string, donc ok pour le format. Semble être des catégories, donc nombre limité de chaines de caractères différentes.
 2   primaryTitle    object --> censé être un string, donc ok pour le format
 3   originalTitle   object --> censé être un string, donc ok pour le format
 4   isAdult         object --> censé être un booléen (0 ou 1), donc MAUVAIS FORMAT
 5   startYear       object --> censé être une date (datetime), donc MAUVAIS FORMAT
 6   endYear         object --> censé être une date (datetime), donc MAUVAIS FORMAT
 7   runtimeMinutes  object --> censé être une durée (datetime (à vérifier)), donc MAUVAIS FORMAT
 8   genres          object --> contient une chaine caractères (string) qui représente une succession d'éléments séparés par des virgules. MAUVAIS FRMAT POUR UN EXPLODE, bon format pour un SPLIT

 Infos : lorsque les données d'une colonne sont sous forme de chaînes de caractères délimitées (comme "1,2,3"), une combinaison avec `.str.split(delimiter)` est nécessaire pour transformer ces chaînes en listes exploitables par `explode()`. Ensuite, pour homogénéiser les données ou gérer les valeurs manquantes, `.fillna()` peut être utilisé pour remplacer les cellules vides par des listes vides avant d'exploser la colonne. Dans des cas où les éléments des listes doivent être convertis dans un type spécifique, comme des entiers ou des dates, `.apply()` ou `.astype()` est utile après le traitement initial.

In [ ]:
# Convertir les formats en datetime : 
df['startYear'] = pd.to_datetime(df['startYear'], format="%Y", errors='raise')
df['startYear'] = pd.to_datetime(df['startYear'], format="%Y", errors='raise')
df['startYear'] = pd.to_datetime(df['startYear'], format="%Y", errors='raise')

Brouillon : 

%Y # année à 4 chiffres
%y # année à 2 chiffres. Ex : '20' pour l'année 2020
%m # mois
%d # jour
%H # heures
%M # minutes